rand_distr/src/binomial.rs

```diff
@@ -210,24 +211,28 @@ impl Distribution<u64> for Binomial {
                     let s = p / q;
                     let a = s * (n + 1.);
                     let mut f = 1.0;
-                    if m < y {
-                        let mut i = m;
-                        loop {
-                            i += 1;
-                            f *= a / (i as f64) - s;
-                            if i == y {
-                                break;
+                    match m.cmp(&y) {
+                        Ordering::Less => {
+                            let mut i = m;
+                            loop {
+                                i += 1;
+                                f *= a / (i as f64) - s;
+                                if i == y {
+                                    break;
+                                }
                             }
-                        }
-                    } else if m > y {
-                        let mut i = y;
-                        loop {
-                            i += 1;
-                            f /= a / (i as f64) - s;
-                            if i == m {
-                                break;
+                        },
+                        Ordering::Greater => {
+                            let mut i = y;
+                            loop {
+                                i += 1;
+                                f /= a / (i as f64) - s;
+                                if i == m {
+                                    break;
+                                }
                             }
-                        }
+                        },
+                        Ordering::Equal => {},
```

Comment:
I think it's worth to double check if this branch is indeed correct.

Reply:
The algorithm as described in the reference does not include a branch for `m == y`.

Reply:
On the other hand, there might be a bug in the specification, I did not check that or compare to the other implementations...